### PR TITLE
tests(smokehouse): snapshot stderr

### PR DESCRIPTION
**Pull request by connorjclark**
adds assertions for the CLI's stdout. They can be updated like this: `yarn smoke -u`

There are a couple of unexpected non-deterministic results, which I'm still looking into.

1. `"config:warn is-on-https in 'onlyAudits' is already included by best-practices in 'onlyCategories'"` varies from `http://localhost:10200/online-only.html`
1. `"GatherRunner:error Lighthouse was unable to reliably load the page you requested. Make sure you are testing the correct URL and that the server is properly responding to all requests. (Status code: 403) http://localhost:10200/seo/seo-failure-cases.html?status_code=403"` varies from `http://localhost:10200/seo/seo-failure-cases.html?extra_header...`

### CODE REVIEW

File: lighthouse-cli/test/smokehouse/run-smoke.js [hunk 1]
```diff
@@ -106,7 +107,7 @@ async function cli() {
   server.listen(10200, 'localhost');
   serverForOffline.listen(10503, 'localhost');
 
-  const argv = process.argv.slice(2);
+  const argv = process.argv.slice(2).filter(arg => !arg.startsWith('-'));
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/run-smoke.js]: These are the most minimal changes. Proper arg parsing is done in #9676

File: lighthouse-cli/test/smokehouse/smokehouse.js [hunk 2]
```diff
@@ -83,7 +83,7 @@ function runLighthouse(url, configPath, isDebug) {
 
     runCount++;
     console.log(`${log.dim}$ ${command} ${args.join(' ')} ${log.reset}`);
-    runResults = spawnSync(command, args, {encoding: 'utf8', stdio: ['pipe', 'pipe', 'inherit']});
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/smokehouse.js]: This seems unnecessary. Also prevents the `DEBUG` printing (few lines later) from working

**Comment by patrickhulce** [comment]
It feels like I'm missing context for why this was done :)

Was there a specific bug of class of bugs we made that this is intended to catch?

**Comment by connorjclark** [comment]
Calls to `log.warn` don't get noticed w/o looking at the stderr. The instigator was that warning that spits out when the iframes gatherer goes unused, however @brendankenny pointed out that was intentional (#9721). however however, being able to tell when calls to log.warn occur / stop occurring seems it could prevent a regression.

**Comment by patrickhulce** [comment]
> being able to tell when calls to log.warn occur / stop occurring seems it could prevent a regression.

how would you feel about doing this in a way that only catches warning and error logs then? :) I'm a bit concerned about the level of churn for logging *every* status statement

**Comment by brendankenny** [comment]
> I'm a bit concerned about the level of churn for logging _every_ status statement

yeah, unless there's a specific bug (or class of bugs) we've been missing, sample_v2 and sample_v2_round_trip (and en-US and en-XL) already seem like enough churn in an average PR. Every snapshot update requires attention to at least evaluate whether or not it's a bug, worth tracking down the root cause of a change, etc.

And logging is in the end really for debugging/human notifications. If there are concerning things only being logged, ideally we should raise them up to `runWarnings`, assertions, etc.

**Comment by brendankenny** [comment]
let's not do this. We don't have a positive result in mind (beyond a general idea of belt and suspenders/braces testing), some of these are going to be difficult to maintain long term and difficult to debug if they disappear (e.g. `method <= browser ERR:error Network.getResponseBody`) and most of the others seem well covered by unit tests. Meanwhile the costs:
- any internal changes or refactors can incur authoring and review cost for things we usually treat as a casual "FYI"
- we add a significant cost if any warnings are flakey, e.g. due to performance differences of the test machine (like suggested in https://github.com/GoogleChrome/lighthouse/issues/9085#issuecomment-535178985), increasing complexity and future maintenance (and annoyance) to keep tests on one side or the other of a flake
- we make things harder for new contributors, one more incantation they need to do to just add some simple change (e.g. [`yarn update:sample-json`](https://github.com/GoogleChrome/lighthouse/pulls?q=is%3Apr+in%3Acomments+%22update%3Asample-json%22+is%3Aclosed))

I think we're better off leaving them as is.

**Pull request closed without merging**